core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -994,10 +994,12 @@ private void listStatusInternal(LockedInodePath currInodePath,
           ? DescendantType.ALL : DescendantType.NONE;
       for (Inode<?> child : ((InodeDirectory) inode).getChildren()) {
         // TODO(david): Make extending InodePath more efficient
-        try (LockedInodePath childInodePath = mInodeTree.lockFullInodePath(child.getId(),
-            InodeTree.LockMode.READ)) {
+        try (LockedInodePath childInodePath = mInodeTree.lockDescendantPath(currInodePath,
+            InodeTree.LockMode.READ, currInodePath.getUri().join(child.getName())))  {
           listStatusInternal(childInodePath, auditContext,
               nextDescendantType, statusList);
+        } catch (InvalidPathException e) {
+          LOG.warn("ListStatus encountered an invalid path {}", child.getName(), e);
```

Comment:
This message should include `currInodePath`, and the `child.getName`, so we can know which path is invalid.

Reply:
done